lib/hanami/configuration.rb

```diff
@@ -42,12 +42,22 @@ def initialize(env:)
       self.root = Dir.pwd
       self.settings_store = Application::Settings::DotenvStore.new.with_dotenv_loaded
 
+      @assets = begin
+        require_path = "hanami/assets/application_configuration"
+        require require_path
+        Hanami::Assets::ApplicationConfiguration.new
+      rescue LoadError => e
+        raise e unless e.path == require_path
+        require_relative "configuration/null_configuration"
+        NullConfiguration.new
+      end
+
       # Config for actions (same for views, below) may not be available if the gem isn't
       # loaded; fall back to a null config object if it's missing
       @actions = begin
         require_path = "hanami/action/application_configuration"
         require require_path
-        Hanami::Action::ApplicationConfiguration.new
+        Hanami::Action::ApplicationConfiguration.new(assets_server_url: assets.server_url)
```

Comment:
@jodosha I think passing the `server_url` value direct here in `#initialize` will be problematic if a user (either manually or via some plugin eventually) ever _changes_ the value of this `server_url` setting away from its default. In that case, the `Action::AppConfiguration` will have a _stale value_ for `server_url`.

I think we'll need to support this as part of `actions.finalize!` (at which point the actions config will know the `server_url` shouldn't change again) or find some other approach.

Are you OK to have a look at this?

Reply:
@timriley Will do, thanks for the good catch. 👍

Reply:
I suspect there'll be other externally-provided config objects that end up with a similar requirement. We may want to consider updating those `finalize!` methods to receive `self` or something like that...